### PR TITLE
Migrated to Typescript

### PR DESCRIPTION
**Pull request by ludohenin**
* `js` files migrated to `ts`.
* gulp file updated, gulp-traceur no more required, traceur-runtime taken from angular2 pkg.
* `.d.ts` file added (angular2.d.ts is a customized one found on github somewhere, to be updated as needed)

### CODE REVIEW

**Comment by johnjelinek** [comment]
@ludohenin: can you squash these into a single commit?

File: ts/app.ts [hunk 1]
```diff
@@ -1,9 +1,5 @@
-import {
-  ComponentAnnotation as Component,
-  ViewAnnotation as View,
-  bootstrap,
-  For
-} from 'angular2/angular2';
+/// <reference path="../typings/angular2/angular2.d.ts" />
+import {Component, View, bootstrap, For} from 'angular2/angular2';
```
Comment (johnjelinek) [inline ts/app.ts]: Why does this work? Shouldn't it still be `ComponentAnnotation as Component` / `ViewAnnotation as View`?

Comment (ludohenin) [inline ts/app.ts]: `ComponentAnnotation` is a traceur annotation. Works how it is supposed to with `ts`.

Comment (ludohenin) [inline ts/app.ts]: Don't know how to squash into a single commit. You can still clone my fork if you can't wait @mgechev to merge

**Comment by johnjelinek** [comment]
`git rebase -i HEAD~2` -- save that and then `git push -f` to your repo.

**Comment by ludohenin** [comment]
by saving you mean ? (sry I'm very new to git and can only handle simple things for now).

**Comment by johnjelinek** [comment]
after you do `git rebase -i HEAD~2` it'll present you with your commit screen, mark the ones you want with a `squash` instead of `pick`. And save/close in your editor. So, for example, in vim you'd do `:wq`. More info [here](http://stackoverflow.com/a/5189600/979315) or [here](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/)

**Comment by ludohenin** [comment]
awesome, thanks bud. Commits squashed :-)

**Comment by johnjelinek** [comment]
:+1:

**Comment by mgechev** [comment]
@ludohenin thanks looks great! There's no source maps support right now. I'll open an issue for this.

**Pull request merged**